src/pages/AddItem.js

```diff
@@ -14,24 +34,62 @@ export default function AddItem(props) {
     setPurchaseFrequency(e.target.value);
   };
 
-  const createListItem = (e) => {
-    e.preventDefault();
-    db.collection('generated_token')
-      .add({
-        item_name: itemName,
-        purchase_frequency: parseInt(purchaseFrequency),
-        last_purchased: lastPurchased,
-      })
-      .then((documentReference) => {
-        console.log('document reference ID', documentReference.id);
-      })
-      .catch((error) => {
-        console.log(error.message);
-      });
-    setItemName('');
-    setLastPurchased(null);
+  const checkForDuplicates = () => {
+    // we would like to potentially normalize white space but unsuccessfull so far
+    const normalizedUserInput = itemName
+      .toLowerCase()
+      .replace(/[^\w\s]|_/g, '');
+
+    const matchingItemName = listItems.filter((item) => {
+      const normalizedDataBaseItem = item.item_name
+        .toLowerCase()
+        .replace(/[^\w\s]|_/g, '');
+      console.log(normalizedDataBaseItem);
+      return normalizedDataBaseItem === normalizedUserInput;
+    });
+
+    if (matchingItemName.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
   };
 
+  async function createListItem(e) {
+    e.preventDefault();
+    try {
+      const result = await checkForDuplicates(itemName);
+      if (result) {
```

Comment:
```suggestion
      if (checkForDuplicates(itemName)) {
```

Reply:
I don't think you need the `await` here since `checkForDuplicates` wasn't defined as `async`.

Reply:
We cleaned up `createListItem` a lot. One of the things we did is remove the `async` and `await` and several of the unused error messages (that either we had commented out, or were just hanging out doing nothing).